### ISSUE TITLE
insert into values , master gangcontext leak

### ISSUE THREAD

**Issue by wjzeng** [greenplum-db/gpdb#2738]
HI team

I found that a simple insert statement would cause gpdb master memleak

Greenplum Database 4.3

This is a detailed step

# table define
```
create sequence id_seq;

CREATE TABLE t_leak (
    id bigint DEFAULT nextval('id_seq'::regclass) NOT NULL,
    id1 character varying(32) DEFAULT ''::character varying,
    id2 bigint,
    id3 character varying(1024) DEFAULT ''::character varying,
    Id4 character varying(1024) DEFAULT ''::character varying,
    Id5 character varying(1024) DEFAULT ''::character varying
)
WITH (appendonly=true, compresstype=zlib, compresslevel=5) DISTRIBUTED BY (id);
```

# SQL
```
runsql:

import psycopg2
import time

connection = psycopg2.connect('user=wenjing.zwj host=1.1.1.1 port=15432 dbname=postgres password=pgsql')
connection.autocommit = True
cursor = connection.cursor()
cursor.execute('''set gp_vmem_idle_resource_timeout=0''')
for i in range(1, 10000000):
    try:
        cursor.execute('''BEGIN''')
        cursor.execute('''insert into t_leak (id1,id2,id3,id4,id5) VALUES ('1.1.1.1','1','android','video on line time','1881')''')
        cursor.execute('''COMMIT''')
        cursor.execute('''insert into t_leak (id1,id2,id3,id4,id5) VALUES ('1.1.1.1','1','android','video on line time','1881')''')
    except Exception, e:
        print '%s' % e
```

# Gang Context    memleak info

```
Gang Context: 128752
 MessageContext: 131072
  UdpInterconnectMemContext: 8192
   Local Stat Portal Hash: 8192
    smgr relation table: 24576
     TransactionAbortContext: 32768
      Portal hash: 8192
       PortalMemory: 8192
        Relcache by OID: 24576
         CacheMemoryContext: 801488
          MdSmgr: 8192
           LOCALLOCK hash: 24576
            Timezones: 83512
             Postmaster: 24576
              MemoryAccountMemoryContext: 8192
               ErrorContext: 8192
              ErrorContext: 8192
             MemoryAccountMemoryContext: 8192
              ErrorContext: 8192
             ErrorContext: 8192
            Postmaster: 24576
             MemoryAccountMemoryContext: 8192
              ErrorContext: 8192
             ErrorContext: 8192
            MemoryAccountMemoryContext: 8192
             ErrorContext: 8192
            ErrorContext: 8192
           Timezones: 83512
            Postmaster: 24576
             MemoryAccountMemoryContext: 8192
              ErrorContext: 8192
             ErrorContext: 8192
            MemoryAccountMemoryContext: 8192
             ErrorContext: 8192
            ErrorContext: 8192
           Postmaster: 24576
            MemoryAccountMemoryContext: 8192
             ErrorContext: 8192
            ErrorContext: 8192
           MemoryAccountMemoryContext: 8192
            ErrorContext: 8192
           ErrorContext: 8192
          LOCALLOCK hash: 24576
           Timezones: 83512
            Postmaster: 24576
             MemoryAccountMemoryContext: 8192
              ErrorContext: 8192
             ErrorContext: 8192
            MemoryAccountMemoryContext: 8192
```

**Comment by hsyuan** [comment]
@xiong-gang @eurekaka

**Comment by eurekaka** [comment]
@hsyuan thx

**Comment by eurekaka** [comment]
Already fixed in latest 4.3 and open source repo.
